src/protocol/RedemptionHandler.sol

```diff
@@ -81,43 +96,75 @@ contract RedemptionHandler is CoreOwnable{
     /// @notice Calculates the total redemption fee as a percentage of the redemption amount.
     function getRedemptionFeePct(address _pair, uint256 _amount) public view returns(uint256){
         //get fee
-        (uint256 feePct,) = _getRedemptionFee(_pair, _amount);
+        (uint256 feePct,,) = _getRedemptionFee(_pair, _amount);
         return feePct;
     }
 
-    function _getRedemptionFee(address _pair, uint256 _amount) internal view returns(uint256, RedeemptionRateInfo memory){
+    function _getRedemptionFee(address _pair, uint256 _amount) internal view returns(uint256 _redemptionfee, RedeemptionRateInfo memory _rdata, uint256 _totalweight){
         (, , , IResupplyPair.VaultAccount memory _totalBorrow) = IResupplyPair(_pair).previewAddInterest();
         
         //determine the weight of this current redemption by dividing by pair's total borrow
         uint256 weightOfRedeem;
         if (_totalBorrow.amount != 0) weightOfRedeem = _amount * PRECISION / _totalBorrow.amount;
 
         //update current data with decay rate
-        RedeemptionRateInfo memory rdata = ratingData[_pair];
+        // RedeemptionRateInfo memory rdata = ratingData[_pair];
+        _rdata = ratingData[_pair];
         
+        _totalweight = totalWeight;
+
         //only decay if this pair has been used before
-        if(rdata.timestamp != 0){
+        if(_rdata.timestamp != 0){
+            //remove current from total (add back after calcs)
+            _totalweight -= _rdata.usage;
```

Comment:
how does `totalWeight` state var get set initially? if starting at 0, this would underflow

Reply:
If rdata is set then totalWeight should also be set